### PR TITLE
[Impeller] Use impellerc for engine FragmentProgram unit tests

### PR DESCRIPTION
**Pull request by zanderso**
Since we're shipping `impellerc` to the framework and using it to compile shaders there, that's what we should be using to run the unit tests in the engine repo.

### CODE REVIEW

File: impeller/compiler/compiler.cc [hunk 1]
```diff
@@ -139,7 +139,7 @@ Compiler::Compiler(const fml::Mapping& source_mapping,
       break;
     case TargetPlatform::kFlutterSPIRV:
       spirv_options.SetOptimizationLevel(
-          shaderc_optimization_level::shaderc_optimization_level_size);
+          shaderc_optimization_level::shaderc_optimization_level_zero);
```
Comment (chinmaygarde) [inline impeller/compiler/compiler.cc:142]: Curious, why?

File: lib/spirv/test/BUILD.gn [hunk 2]
```diff
@@ -15,12 +15,4 @@ if (enable_unittests) {
       "//third_party/swiftshader_flutter:spvtools_val",
     ]
   }
-
-  executable("glsl_to_spirv") {
```
Comment (chinmaygarde) [inline lib/spirv/test/BUILD.gn:19]: Why can't this entire GN file be removed?

**Review by chinmaygarde — APPROVED** [review]
Some nits but otherwise LGTM.

File: impeller/compiler/compiler.cc (hunk 1 above)
Comment (zanderso) [reply impeller/compiler/compiler.cc:148]: The transpiler is missing some ops: https://github.com/flutter/flutter/issues/105396. Added a comment.

Also it's the level being used by the current test harness: https://github.com/flutter/engine/blob/d0ec2205b34e89f4f8c887d1d378abad91ed9df9/lib/spirv/test/glsl_to_spirv.cc#L39

File: lib/spirv/test/BUILD.gn (hunk 2 above)
Comment (zanderso) [reply lib/spirv/test/BUILD.gn:19]: The [program above](https://github.com/flutter/engine/blob/d0ec2205b34e89f4f8c887d1d378abad91ed9df9/lib/spirv/test/spirv_assembler.cc) is still used. It looks like it is used for assembling invalid bytecode to make sure that the transpiler throws errors correctly.

**Pull request merged**